gelbooru-overhaul.configManager.js

```diff
@@ -0,0 +1,462 @@
+    /**
+     * @param {string} prefKey 
+     * @returns {PreferenceItem} Found preference item or undefined
+     */
+    findItemByKey(prefKey){
+        if (!prefKey || prefKey.length < 1) return undefined;
+        let args = prefKey.split(".");
+
+        let cat = this.config[args[0]];
+        if (!cat) return undefined;
+
+        let item = cat.items[args[1]];
+        if (!item) return undefined;
+
+        return item;
+    }
+    /**
+     * @param {string} prefKey
+     * @returns {string | number | boolean} Found preference value or undefined
+     */
+    findValueByKey(prefKey){
+        let item = this.findItemByKey(prefKey);
+        if (!item) return undefined;
+        
+        return item.value;
+    }
+    /**
+     * Load config from the userscrip manager storage.
+     * 
+     * If there is no saved config then load default.
+     * 
+     * If saved config is outdated then make migrations.
+     * @method loadConfig
+     */
+    loadConfig() {
+        let cfg = GM_getValue("config", undefined);
+
+        let unpackPreferences = this.unpackPreferences(cfg);
+        this.config = unpackPreferences.preferences;
+        if (unpackPreferences.isShouldBeSaved) this.saveConfig();
+
+        utils.debugLog("Loaded config", this.config);
+    }
+    /**
+     * Saves current config to the userscript storage
+     * @method saveConfig
+     */
+    saveConfig() {
+        GM_setValue("config", this.packPreferences(this.config));
+    }
+    /**
+     * Applies current config values
+     * @method applyConfig
+     */
+    applyConfig() {
+        Object.entries(this.config).forEach(ce => {
+            Object.entries(ce[1].items).forEach(ie => {
+                this.onUpdate(`${ce[0]}.${ie[0]}`, ie[1].value);
+            });
+        });
+    }
+    /**
+     * Clears config storage with default config
+     * @method setDefaultConfig
+     */
+    setDefaultConfig() {
+        this.config = this.getDefaultConfig();
+        GM_setValue("config", {});
+        this.saveConfig();
+    }
+    /**
+     * Get default config with current configVersion
+     * @private
+     * @method getDefaultConfig
+     * @returns {Object.<string, PreferenceCategory>} Default config with current configVersion
+     */
+    getDefaultConfig() {
+        return {
+            general: {
+                name: "General",
+                items: {
+                    configVersion: {
+                        value: 1,
+                        name: "Config Version",
+                        description: "Debug info of current config version",
+                        locked: true
+                    },
+                    debug: {
+                        value: false,
+                        name: "Debug",
+                        description: "Enable debugging information in the console",
+                    },
+                    maxCache: {
+                        value: 420,
+                        name: "Max cahed post items",
```

Comment:
```suggestion
                        name: "Max cached post items",
```
Minor spelling mistake.

Reply:
In fact, there are a lot of *Engrish*
When I finished, I was going to ask you to check and correct the visible text